Update coincidence-of-wants.md (#303)
cowprotocol/docs: docs/cow-protocol/concepts/how-it-works/coincidence-of-wants.md

# Coincidence of Wants

> **CoW (Coincidence of Wants)**: An economic phenomenon where two parties — each holding the asset that the other needs — exchange assets directly in an equivalent barter. 

Reviewer: ```suggestion
**CoW (Coincidence of Wants)**: An economic phenomenon where two parties - each holding the asset that the other needs - exchange assets directly in an equivalent barter. 
```

No use of em-dash within markdown source code.

A simple CoW is where two traders are each selling what the other is buying.
They swap assets with each other instead of interacting with AMM liquidity pools.
Simple CoWs can be either *complete*, where each trader is trading exactly the same amount that the other needs, or they can be *partial* where two traders are trading opposite assets but only one can have their full liquidity met through a CoW.

> In the latter case, the rest of the trade goes to an AMM to finish filling. 

Reviewer: This is not completely true as the difference for the CoW may come from a private market maker as well. It would be more correct to refer to `on-chain` here (of which would be AMM/PMM etc).

Author: So wouldn't it be more correct to say "on-chain" (as we have it) than "to an AMM"? In your suggestion you're restricting it to just "AMM" rather than the more broad "on-chain"

> In the latter case, the rest of the trade goes to an AMM to finish filling. 

Reviewer: ```suggestion
In the latter case, the rest of the trade goes on-chain to finish filling. 
```

### Batching

![Batching CoW](/img/concepts/batching-cow.png)

Often, a batch auction contains multiple intents from different users who all want to perform the same trade.
Instead of sending their trades individually one by one, solvers can batch all of the intents together and execute them as a single trade on-chain.

> Although a batching CoW goes to an AMM, this optimization reduces gas fees, as users trades interact with AMM smart contracts fewer times.

Reviewer: Also not quite true (refer above)

> Although a batching CoW goes to an AMM, this optimization reduces gas fees, as users trades interact with AMM smart contracts fewer times.

Reviewer: ```suggestion
Although a batching CoW goes on-chain, this optimization reduces gas fees, as users trades interact with AMM smart contracts fewer times.
```

It often includes multiple assets and relies on the solvers to find the optimal path from each user to the next in order to maximize off-chain liquidity utilization. 

While some orders are filled entirely through a CoW, many are only partially filled peer-to-peer.

> In these latter cases, solvers automatically go to an AMM to source the rest of the liquidity necessary to finish filling the order.

Reviewer: As above

> In these latter cases, solvers automatically go to an AMM to source the rest of the liquidity necessary to finish filling the order.

Reviewer: ```suggestion
In these latter cases, solvers automatically go on-chain to source the rest of the liquidity necessary to finish filling the order.
```
